### PR TITLE
py-contextlib2: Add Python 39, rm 35 36

### PR DESCRIPTION
**Pull request by essandess**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by herbygillot** [comment]
Since you are removing the subports for Python 3.5 & 3.6, did you check for ports which depend on `py-contextlib2` but still have subports for those versions?  Those will break if this is committed.

**Comment by essandess** [comment]
> Since you are removing the subports for Python 3.5 & 3.6, did you check for _other_ ports which depend on `py-contextlib2` but still have subports for those versions? Those will break if this is committed.

I did not. Is there a one-liner that accomplishes this without having the whole upstream DAG installed?

Also, 35 has been EOL since last year. I can see the argument for keeping some legacy 27 code active, but it’s time to move on from 35. 36 hits EOL in a few months too.

**Comment by essandess** [comment]
> I did not. Is there a one-liner that accomplishes this without having the whole upstream DAG installed?

Pull the plug?
* https://ports.macports.org/port/py35-contextlib2/summary
* https://ports.macports.org/port/py36-contextlib2/summary

File: python/py-contextlib2/Portfile [hunk 1]
```diff
@@ -11,12 +11,12 @@ platforms           darwin
 license             PSF
 supported_archs     noarch
 
-python.versions     27 35 36 37 38
+python.versions     27 37 38 39
```
Comment (reneeotten) [inline python/py-contextlib2/Portfile:14]: don't remove subports that still have dependents:
```
% port echo depends:py35-contextlib2
py35-cherrypy

% port echo depends:py36-contextlib2
py36-cherrypy
```

**Comment by herbygillot** [comment]
> > I did not. Is there a one-liner that accomplishes this without having the whole upstream DAG installed?
> 
> Pull the plug?
> 
> * https://ports.macports.org/port/py35-contextlib2/summary
> * https://ports.macports.org/port/py36-contextlib2/summary

No reason to pull the plug, still add 3.9 support, just don't remove 3.5 and 3.6 for now.

**Comment by reneeotten** [comment]
> I did not. Is there a one-liner that accomplishes this without having the whole upstream DAG installed?

it seems I missed part of the conversation when I reviewed this... but yes, you can/should check this without the need to install anything using: ```port echo depends:py35-contextlib2```

> Also, 35 has been EOL since last year. I can see the argument for keeping some legacy 27 code active, but it’s time to move on from 35. 36 hits EOL in a few months too.

I have no issues with removing EOL subports (i.e., py35 for now), it's just that if you want to do that you have to do this for all dependencies recursively so that nothing breaks..

**Comment by cjones051073** [comment]
closing, as added py39 independently.

**Pull request closed without merging**